File: Platform/benchmark/urls.py
from django.urls import path
from benchmark import views

app_name = 'benchmark'

urlpatterns = [
    # ==========================================
    # 1. Views (HTML Pages)
    # ==========================================
    path("", views.home, name="home"),
    path("vanilla_llm/", views.vanilla_llm, name="vanilla_llm"),
    path("rag/", views.rag, name="rag"),
    path("vanilla_agent/", views.vanilla_agent, name="vanilla_agent"),
    path("browser_agent/", views.browser_agent, name="browser_agent"),

    # ==========================================
    # 2. API - Common / General
    # ==========================================
    path('api/web_search/', views.web_search, name='web_search'),

    # ==========================================
    # 3. API - Global Settings
    # ==========================================
    path("api/get_default_settings/", views.get_default_settings, name="get_default_settings"),
    path("api/settings/save/", views.save_settings, name="save_settings"),
    path("api/test_llm_connection/", views.test_llm_connection, name="test_llm_connection"),

    # ==========================================
    # 3.1. API - Metrics
    # ==========================================
    path("api/metrics/calculate/", views.calculate_metrics, name="calculate_metrics"),
    path("api/metrics/colors/", views.get_metric_colors, name="get_metric_colors"),
    path("api/metrics/schema/", views.get_metric_schema, name="get_metric_schema"),

    # ==========================================
    # 4. API - Datasets
    # ==========================================
    path('api/datasets/list/', views.dataset_list, name='dataset_list'),
    path('api/datasets/<int:dataset_id>/questions/', views.get_dataset_questions, name='get_dataset_questions'),
    path('api/datasets/upload/', views.dataset_upload, name='dataset_upload'),
    path('api/datasets/delete/<int:dataset_id>/', views.dataset_delete, name='dataset_delete'),
    path('api/datasets/sync/', views.sync_datasets, name='sync_datasets'),
    path('api/datasets/activate/<int:dataset_id>/', views.activate_dataset, name='activate_dataset'),

    # ==========================================
    # 5. API - Multi-Turn Sessions
    # ==========================================
    # Session Management
    path("api/sessions/create_session_group/", views.create_session_group, name="create_session_group"),
    path("api/sessions/create_session/", views.create_session, name="create_session"),
    path("api/sessions/get_session/<int:session_id>/", views.get_session, name="get_session"),
    path("api/sessions/delete_session/<int:session_id>/", views.delete_session, name="delete_session"),
    path('api/sessions/batch_delete_sessions/', views.batch_delete_sessions, name='batch_delete_sessions'),
    path('api/sessions/delete_session_group/<int:group_id>/', views.delete_session_group, name='delete_session_group'),
    path('api/sessions/rename_session_group/<int:group_id>/', views.rename_session_group, name='rename_session_group'),
    path('api/sessions/export_session/<int:session_id>/', views.export_session, name='export_session'),
    path('api/sessions/export_run/<int:group_id>/', views.export_run, name='export_run'),
    path('api/sessions/import/', views.import_data, name='import_data'),
    path('api/sessions/validate_import/', views.validate_import, name='validate_import'),
    
    # Trial Execution
    path("api/sessions/run_trial/<int:trial_id>/", views.run_trial, name="run_trial"),
    path("api/sessions/retry_session/<int:trial_id>/", views.retry_session, name="retry_session"),
    path("api/sessions/stop_session/", views.stop_session, name="stop_session"),
    path("api/sessions/get_trial_trace/<int:trial_id>/", views.get_trial_trace, name="get_trial_trace"),
    path("api/sessions/get_trial_prompt/<int:trial_id>/", views.get_trial_prompt, name="get_trial_prompt"),

    # Run Loading
    path('api/sessions/load_vanilla_run/<int:group_id>/', views.load_benchmark_run, {'pipeline_category': 'vanilla_llm'}, name='load_vanilla_llm_run'),
    path('api/sessions/load_rag_run/<int:group_id>/', views.load_benchmark_run, {'pipeline_category': 'rag'}, name='load_rag_run'),
    path('api/sessions/load_agent_run/<int:group_id>/', views.load_benchmark_run, {'pipeline_category': 'agent_multi_turn'}, name='load_agent_multi_turn_run'),

    # ==========================================
    # 6. API - Pipelines (Streaming)
    # ==========================================
    path('api/pipeline/start/<str:pipeline_type>/', views.pipeline_start, name='pipeline_start'),
    path('api/pipeline/stop/<str:pipeline_type>/', views.pipeline_stop, name='pipeline_stop'),
]

File: Platform/static/js/benchmark/config/urls.js
/**
 * Centralized URL configuration for Benchmark API endpoints
 */

const API_PREFIX = '/benchmark/api';

window.BenchmarkUrls = {
    // LLM & Settings
    saveSettings: `${API_PREFIX}/settings/save/`,
    getDefaultSettings: `${API_PREFIX}/get_default_settings/`,
    testLlmConnection: `${API_PREFIX}/test_llm_connection/`,

    // API - Common / General
    webSearch: `${API_PREFIX}/web_search/`,

    // Metrics
    metrics: {
        calculate: `${API_PREFIX}/metrics/calculate/`,
        colors: `${API_PREFIX}/metrics/colors/`,
        schema: `${API_PREFIX}/metrics/schema/`
    },

    // Leaderboard
    leaderboard: {
        get: `${API_PREFIX}/leaderboard/`,
        getFiltered: (params) => {
            const queryString = new URLSearchParams(params).toString();
            return `${API_PREFIX}/leaderboard/${queryString ? '?' + queryString : ''}`;
        }
    },

    // Datasets
    datasets: {
        sync: `${API_PREFIX}/datasets/sync/`,
        upload: `${API_PREFIX}/datasets/upload/`,
        delete: (id) => `${API_PREFIX}/datasets/delete/${id}/`,
        activate: (id) => `${API_PREFIX}/datasets/activate/${id}/`
    },

    // Multi-turn Common
    multiTurn: {
        createSession: `${API_PREFIX}/sessions/create_session/`,
        createSessionGroup: `${API_PREFIX}/sessions/create_session_group/`,
        batchDeleteSessions: `${API_PREFIX}/sessions/batch_delete_sessions/`,
        deleteSessionGroup: (id) => `${API_PREFIX}/sessions/delete_session_group/${id}/`,
        renameSessionGroup: (id) => `${API_PREFIX}/sessions/rename_session_group/${id}/`,
        getSession: (id) => `${API_PREFIX}/sessions/get_session/${id}/`,
        runTrial: (id) => `${API_PREFIX}/sessions/run_trial/${id}/`,
        retrySession: (id) => `${API_PREFIX}/sessions/retry_session/${id}/`,
        deleteSession: (id) => `${API_PREFIX}/sessions/delete_session/${id}/`,
        stopSession: `${API_PREFIX}/sessions/stop_session/`,
        exportSession: (id) => `${API_PREFIX}/sessions/export_session/${id}/`,
        exportRun: (id) => `${API_PREFIX}/sessions/export_run/${id}/`,
        getTrialPrompt: (id) => `${API_PREFIX}/sessions/get_trial_prompt/${id}/`,
    },

    // Pipelines (Unified)
    pipeline: {
        start: (type) => `${API_PREFIX}/pipeline/start/${type}/`,
        stop: (type) => `${API_PREFIX}/pipeline/stop/${type}/`,
        loadRun: (type, id) => {
            const map = {
                'vanilla_llm': 'load_vanilla_run',
                'rag': 'load_rag_run',
                'vanilla_agent': 'load_agent_run',
                'browser_agent': 'load_agent_run'
            };
            return `${API_PREFIX}/sessions/${map[type] || 'load_vanilla_run'}/${id}/`;
        }
    },

    // Compatibility (Mapping to unified)
    vanillaLlmMultiTurn: {
        loadRun: (id) => `${API_PREFIX}/sessions/load_vanilla_run/${id}/`,
        runPipeline: `${API_PREFIX}/pipeline/start/vanilla_llm/`,
        stopPipeline: `${API_PREFIX}/pipeline/stop/vanilla_llm/`
    },
    ragMultiTurn: {
        loadRun: (id) => `${API_PREFIX}/sessions/load_rag_run/${id}/`,
        runPipeline: `${API_PREFIX}/pipeline/start/rag/`,
        stopPipeline: `${API_PREFIX}/pipeline/stop/rag/`
    },
    vanillaAgent: {
        loadRun: (id) => `${API_PREFIX}/sessions/load_agent_run/${id}/`,
        runPipeline: `${API_PREFIX}/pipeline/start/vanilla_agent/`,
        stopPipeline: `${API_PREFIX}/pipeline/stop/vanilla_agent/`
    },
    browserAgent: {
        loadRun: (id) => `${API_PREFIX}/sessions/load_agent_run/${id}/`,
        runPipeline: `${API_PREFIX}/pipeline/start/browser_agent/`,
        stopPipeline: `${API_PREFIX}/pipeline/stop/browser_agent/`
    }
};
